### DISCUSSION TITLE
Authentication failed with API key

### DISCUSSION THREAD

**Discussion by garnerh42** [robertodoering/harpy#886] [Q&A]
I just signed up today for my custom API key and entered the data into the app.  I clicked on "setup guide" and then "login with twitter" and got "authentication failed, please try again".  I am fairly sure I entered in the correct data.  I went the Twitter developer page and clicked on "reveal api key hint" to see if the last few letters matched.

Perhaps I need to wait a few hours for it to work?  Did I miss a step?

**Comment by robertodoering** [comment]
> I clicked on "setup guide" and then "login with twitter"

Did you follow the setup guide?

> Perhaps I need to wait a few hours for it to work?

You don't need to wait any time. If you get "authentication failed" you have either entered the wrong key or secret or you have not set up the key correctly.

**Reply by garnerh42** [reply]
I recreated the API keys, the Bearer Token and the Access Token/Secret.  The latter I did after I changed the permissions to Read, Write and Direct Messages.  Still didn't work with Harpy.

I looked up a way to use this data in Python and installed tweepy.  With the Bearer Token I was able to do simple queries which indicates that at least part of my setup is working.  However when I went to initialize with the keys and access tokens I got the 401 Unauthorized error.  At least I can reproduce my issue without the Android app.

Can you offer some suggestions on diagnosing this further or something else to check on the Twitter Dev Portal?  Thanks for your time!

**Reply by robertodoering** [reply]
Did you go into the user authentication settings for your api key on the Twitter developer platform and set it up as described in the setup guide?

**Reply by garnerh42** [reply]
I went back over the setup instructions and saw that I entered the url for the callback URI.  Once I saved that I was able to get past the login error.  Progress, thanks!

However, I am stuck in a loop of logging in and authorizing to access my account.  I use a password manager so I know I'm entering in my password correctly (it works without my custom API keys).

Any ideas?

**Reply by robertodoering** [reply]
Do you have elevated access to the api?
You can check at https://developer.twitter.com/en/portal/products

**Reply by garnerh42** [reply]
No, I appear to have glossed over that part of the instructions.  My bad.  I will apply and let you know.

**Reply by garnerh42** [reply]
They approved that pretty fast for a weekend.  All is working!  Thank you for your patience.  And that'll teach me to have patience when following directions in the future :)

And thank you for the great app!  Let's hope Elon changes his mind...

**Reply by H-Tevin** [reply]
I got stuck in the same loop, but these messages got me through it. I would've been lost if you hadn't run into the issue too, so thanks for your struggle lol

I just bought pro too. Here's to using Harpy for many years to come. Thanks so much for all your hard work @robertodoering :)

**Reply by robertodoering** [reply]
Nice to know it works now @garnerh42!
And thanks a lot for the support @H-Tevin :)
